File: java/src/main/java/net/dontcode/core/Message.java
package net.dontcode.core;

/**
 * Definition of a message exchanged between services or between typescript client and services
**/

public class Message {

    protected MessageType type;
    protected String sessionId;
    protected Change change;

    public static enum MessageType {
        INIT,
        CHANGE
    }
}

File: node/packages/core/src/lib/message/message.ts
import { Change } from '@dontcode/core';

/**
 * Description of a message exchanged between the components (client, services)
 * It can be an INIT with the sessionId requested or given
 * Or a CHANGE with the sessionId and Change
 */
export class Message {
  type: MessageType;
  sessionId?: string;
  change?: Change;

  constructor(type: MessageType, sessionId?: string, change?: Change) {
    this.type = type;
    this.sessionId = sessionId;
    this.change = change;
  }
}

export enum MessageType {
  INIT = 'INIT',
  CHANGE = 'CHANGE',
}
